File: Go/Set/set_test.go
package main

import (
	"reflect"
	"testing"
)

func TestSet(t *testing.T) {

	set1, set2 := NewSet(1), NewSet(2)

	set1.Add(1)
	set1.Add(3)
	set1.Add(5)

	set2.Add(2)
	set2.Add(4)
	set2.Add(6)

	set1.Remove(5)
	set2.Remove(8)

	set1.Print()
	set2.Print()

	b1 := set1.ToMarshal()
	// fmt.Println("b1: ", b1)
	b2 := set2.ToMarshal()

	rid1, updates1 := FromMarshalData(b1)
	rid2, updates2 := FromMarshalData(b2)

	set3, set4 := NewSet(3), NewSet(4)

	set3.Merge(rid1, updates1)
	set4.Merge(rid2, updates2)

	set3.Print()
	set4.Print()

	if !reflect.DeepEqual(set1.SortedValues(), set3.SortedValues()) {
		t.Errorf("set1 and set3 values are not the same")
	}

	if !reflect.DeepEqual(set2.SortedValues(), set4.SortedValues()) {
		t.Errorf("set2 and set4 values are not the same")
	}

}

File: JS/Set/test/set.test.js
var assert = require('assert') //Use var, const gave error
const SetCRDT = require('../set')

describe('SetCRDT', function () {
    it('should perform operations and maintain correct state', async function () {
        const set1 = new SetCRDT(1)
        const set2 = new SetCRDT(2)

        await set1.add(1)
        await set1.add(3)
        await set1.add(5)

        await set2.add(2)
        await set2.add(4)
        await set2.add(6)

        await set1.remove(5)
        await set2.remove(8)

        set1.print()
        set2.print()


        const b1 = set1.toMarshal()
        // console.log("b1: ", b1)
        const b2 = set2.toMarshal()

        const [rid1, updates1] = SetCRDT.fromMarshalData(b1)
        const [rid2, updates2] = SetCRDT.fromMarshalData(b2)

        const set3 = new SetCRDT(3)
        const set4 = new SetCRDT(4)

        await set3.merge(rid1, updates1)
        await set4.merge(rid2, updates2)

        set3.print()
        set4.print()

        assert.deepStrictEqual(set1.getValue(), set3.getValue(), "set1 and set3 are not equal")
        assert.deepStrictEqual(set2.getValue(), set4.getValue(), "set2 and set4 are not equal")
    })
})